pkg/queue/handlers/http_request.go

```diff
@@ -0,0 +1,258 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/contiamo/go-base/pkg/queue"
+	"github.com/contiamo/go-base/pkg/queue/workers"
+	"github.com/contiamo/go-base/pkg/tokens"
+	"github.com/contiamo/go-base/pkg/tracing"
+	"github.com/opentracing/opentracing-go"
+	otext "github.com/opentracing/opentracing-go/ext"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// HTTPRequestTask marks a task as an HTTP request task
+	HTTPRequestTask queue.TaskType = "http-request"
+)
+
+// HTTPRequestTaskSpec describes the specification of the HTTP request task
+type HTTPRequestTaskSpec struct {
+	// Method to use for the HTTP request
+	Method string `json:"method"`
+	// URL is the target URL for the request.
+	// Must be an absolute URL that contains the scheme and the host components.
+	URL string `json:"url"`
+	// RequestBody to send
+	RequestBody string `json:"requestBody"`
+	// RequestHeaders to send
+	RequestHeaders map[string]string `json:"requestHeaders"`
+	// Authorized if `true` the task will send a header with the
+	// signed JWT token as a part of the request
+	Authorized bool `json:"authorized"`
+	// ExpectedStatus is an HTTP status expected as a response.
+	// If it does not match the actual status the task fails
+	ExpectedStatus int `json:"expectedStatus"`
+}
+
+type HTTPRequestStage string
+
+var (
+	// RequestPreparing means the task is preparing the request parameters and the body
+	RequestPreparing HTTPRequestStage = "preparing"
+	// RequestPending means the request was sent, awaiting the response
+	RequestPending HTTPRequestStage = "pending"
+	// RequestResponse means the response was received
+	RequestResponse HTTPRequestStage = "response"
+)
+
+// HTTPRequestProgress describes the progress of the HTTP request task stored during
+// the heartbeat handling
+type HTTPRequestProgress struct {
+	// Stage is the current stage of the HTTP request task
+	Stage HTTPRequestStage `json:"stage,omitempty"`
+	// Duration of the HTTP request in milliseconds
+	Duration *int64 `json:"duration,omitempty"`
```

Comment:
can this be a time.Duration?

Reply:
you get nice formatting and clearer semantics

Reply:
good point, will do.